website/docs/src/content/component-reference/storage-components/search-reference.mdx

```diff
@@ -0,0 +1,388 @@
+      category: "article",
+      createdAt: "2023-07-15"
+    }
+  },
+  // ...more results
+]
+```
+
+### delete
+
+Deletes vectors by their IDs.
+
+```tsx
+async delete(options: { ids: Id[] }): Promise<void>
+```
+
+#### Parameters
+
+| Parameter | Type | Description |
+|-----------|------|-------------|
+| `ids` | `string[]` | Array of vector IDs to delete |
+
+#### Example
+
+```tsx
+await namespace.delete({
+  ids: ["doc-1", "doc-2"]
+});
+```
+
+### deleteByFilter
+
+Deletes vectors that match the specified filter.
+
+```tsx
+async deleteByFilter(options: { filters: Filters }): Promise<number>
+```
+
+#### Parameters
+
+| Parameter | Type | Description |
+|-----------|------|-------------|
+| `filters` | `Filters` | Filters to match vectors for deletion |
+
+#### Example
+
+```tsx
+const deletedCount = await namespace.deleteByFilter({
+  filters: {
+    where: {
+      category: { $eq: "outdated" }
+    }
+  }
+});
+```
+
+#### Return Value
+
+Returns the number of deleted vectors.
+
+### getSchema
+
+Retrieves the current schema for the namespace.
+
+```tsx
+async getSchema(): Promise<Schema>
+```
+
+#### Example
+
+```tsx
+const schema = await namespace.getSchema();
+console.log(schema);
+// {
+//   text: "string",
+//   category: "string",
+//   createdAt: "string"
+// }
+```
+
+### updateSchema
+
+Updates the schema for the namespace.
+
+```tsx
+async updateSchema(options: { schema: Schema }): Promise<Schema>
+```
+
+#### Parameters
+
+| Parameter | Type | Description |
+|-----------|------|-------------|
+| `schema` | `Schema` | New schema definition |
+
+#### Example
+
+```tsx
+const updatedSchema = await namespace.updateSchema({
+  schema: {
+    text: "string",
+    category: "string",
+    createdAt: "string",
+    newField: "number",   // Add new field
+    tags: "string[]"      // Add array field
+  }
+});
+```
+
+#### Return Value
+
+Returns the updated schema.
+
+### getMetadata
+
+Retrieves metadata about the namespace.
+
+```tsx
+async getMetadata(): Promise<NamespaceMetadata>
+```
+
+#### Example
+
+```tsx
+const metadata = await namespace.getMetadata();
+console.log(metadata);
+// {
+//   vectorCount: 1250,
+//   dimensions: 1536,
+//   distanceMetric: "cosine_distance",
+//   created: "2023-07-15T12:34:56Z"
+// }
+```
+
+## Namespace Management
+
+Higher-level operations for managing namespaces (these are accessed directly from the search object, not via `useSearch`):
+
+```tsx
+import { SearchClient } from "@gensx/storage";
+
+const search = new SearchClient();
+
+// List all namespaces
+const namespaces = await search.listNamespaces({
+  prefix: "docs-"  // Optional prefix filter
+});
+
+// Check if namespace exists
+const exists = await search.namespaceExists("my-namespace");
+
+// Create namespace if it doesn't exist
+const { created } = await search.ensureNamespace("my-namespace");
+
+// Delete a namespace
+const { deleted } = await search.deleteNamespace("old-namespace");
+```
+
+## Query Result Format
```

Comment:
this can be removed, it's already in the return value section of the query docs above